geospaas_harvesting/crawlers.py

```diff
@@ -233,15 +230,20 @@ def _add_folder_to_process(self, path):
                 self._to_process.append(path)
 
     def _process_folder(self, folder_path):
-        """Get the contents of a folder and feed the _urls and _to_process attributes"""
+        """
+        Get the contents of a folder and feed the _urls (based on includes) and _to_process
+        attributes
+        """
         self.LOGGER.info("Looking for resources in '%s'...", folder_path)
         for path in self._list_folder_contents(folder_path):
-            # Select paths which do not contain any of the self.excludes strings
-            if all(excluded_string not in path for excluded_string in self.excludes):
-                if self._is_folder(path):
-                    self._add_folder_to_process(path)
-                elif self._is_file(path):
-                    self._add_url_to_return(path)
+            # deselect paths which contains any of the excludes strings
+            if self.EXCLUDE and self.EXCLUDE.search(path):
+                continue
+            # select paths which are matched based on input config file
+            if self.include and self.include.search(path):
+                self._add_url_to_return(path)
+            if self._is_folder(path):
+                self._add_folder_to_process(path)
```

Comment:
This should be placed before the exclude test.
As discussed before, folders always need to be processed, otherwise you can't exclude a folder from the output while including the files it contains.

Reply:
Done. the corresponding test is also changed.

Reply:
Done. This way, exclude looses its functionality for folders. Actually, what is my opinion is that the developer should assign the exclude for the folders that we have NEVER EVER need to look inside of it in the online resource. But this way, all folders are will be crawled!

Reply:
How else would you manage the case were you don't want folder paths in the output?

Reply:
Discussed internally.